Packs/SplunkPy/Integrations/SplunkPy/SplunkPy.py

```diff
@@ -497,15 +565,50 @@ def fetch_incidents(service):
 
     oneshotsearch_results = service.jobs.oneshot(searchquery_oneshot, **kwargs_oneshot)  # type: ignore
     reader = results.ResultsReader(oneshotsearch_results)
+
+    last_run_fetched_ids = demisto.getLastRun().get('found_incidents_ids', {})
+
+    current_epoch_time = int(time.time())
+
     for item in reader:
         inc = notable_to_incident(item)
-        incidents.append(inc)
+        opt_in_log('Incident data after parsing to notable: {}'.format(inc))
+        incident_id = create_incident_id(inc)
+
+        if incident_id not in last_run_fetched_ids:
+            last_run_fetched_ids[incident_id] = current_epoch_time
+            incidents.append(inc)
+        else:
+            opt_in_log('SplunkPy - Dropped incident due to duplication.')
+
+    last_run_fetched_ids = replace_dict_comprehension(last_run_fetched_ids, current_epoch_time)
+    opt_in_log('SplunkPy - incidents fetched on last run = {}'.format(last_run_fetched_ids))
+
+    debug_message = 'SplunkPy - total number of incidents found: from {}\n to {}\n with the ' \
+                    'query: {} is: {}.\n incidents found: {}'.format(last_run, now, searchquery_oneshot,
+                                                                     len(incidents), incidents)
+    opt_in_log(debug_message)
+    latest_incident_fetched_time = None if len(incidents) == 0 else get_latest_incident_time(incidents)
+
+    fetches_with_same_start_time_count = demisto.getLastRun().get('fetch_start_update_count', 0) + 1
 
     demisto.incidents(incidents)
-    if len(incidents) < FETCH_LIMIT:
-        demisto.setLastRun({'time': now, 'offset': 0})
+    opt_in_log('SplunkPy - Found incidents at the end of this run: {}'.format(last_run_fetched_ids))
```

Comment:
this are not the incident found in this run, it include incidents from other runs that were not cleaned. is this intended?